src/platform/src/main/java/org/geoserver/platform/resource/Resources.java

```diff
@@ -432,21 +467,216 @@ public static Resource fromPath(String path) {
+            String name = prefix + uuid + suffix;
+            res = dir.get(name);
+        } while(exists(res));
+
+        return res;
+    }
+    
+    
+    /**
+     * Used to look up files based on user provided url (or path).
+     * 
+     * This method (originally from vfny GeoserverDataDirectory) is used to process a URL provided
+     * by a user: <i>iven a path, tries to interpret it as a file into the data directory, or as an absolute
+     * location, and returns the actual absolute location of the file.</i>
+     * 
+     * Over time this url method has grown in the telling to support:
+     * <ul>
+     * <li>Actual URL to external resoruce using http or ftp protocol - will return null</li>
+     * <li>Resource URL - will support resources from resource store</li>
+     * <li>File URL - will support absolute file references</li>
+     * <li>File URL - will support relative file references - this is deprecated, use resource: instead</li>
+     * <li>Fake URLs - sde://user:pass@server:port - will return null.</li>
+     * <li>path - user supplied file path (operating specific specific)</li>
+     * </ul>
+     * 
+     * Note that the baseDirectory is optional (and may be null).
+     * 
+     * @param url File URL or path relative to data directory 
+     * 
+     * @return File indicated by provided URL 
+     */
+    public static Resource fromURL(String path) {
+       return ((GeoServerResourceLoader) GeoServerExtensions.bean("resourceLoader")).fromURL(path);
+    }
+    
+    /**
+     * Used to look up files based on user provided url (or path).
+     * 
+     * This method (originally from vfny GeoserverDataDirectory) is used to process a URL provided
+     * by a user: <i>iven a path, tries to interpret it as a file into the data directory, or as an absolute
+     * location, and returns the actual absolute location of the file.</i>
+     * 
+     * Over time this url method has grown in the telling to support:
+     * <ul>
+     * <li>Actual URL to external resoruce using http or ftp protocol - will return null</li>
+     * <li>Resource URL - will support resources from resource store</li>
+     * <li>File URL - will support absolute file references</li>
+     * <li>File URL - will support relative file references - this is deprecated, use resource: instead</li>
+     * <li>Fake URLs - sde://user:pass@server:port - will return null.</li>
+     * <li>path - user supplied file path (operating specific specific)</li>
+     * </ul>
+     * 
+     * Note that the baseDirectory is optional (and may be null).
+     * 
+     * @param baseDirectory Optional base directory used to resolve relative file URLs
+     * @param url File URL or path relative to data directory 
+     * 
+     * @return File indicated by provided URL 
+     */
+    public static Resource fromURL(Resource baseDirectory, String url) {
+        if (url.startsWith("resource:")) {
+            url = url.substring(9); // remove 'resource:' prefix
```

Comment:
@NielsCharlier I think you inherited this and similar idioms in this PR (perhaps from @jodygarnett?). I dislike hardcoded numeric constants (magic numbers), and I never like to see programmers manually counting string lengths in languages more recent than FORTRAN 66: https://en.wikipedia.org/wiki/Hollerith_constant

```org.apache.commons.lang.StringUtils.removeStart``` may enable a cleaner idiom like:
```
        String ss;
        if ((ss = StringUtils.removeStart(url, "resource:")) != url) {
            return baseDirectory.get(ss);
        }
```
and ```ss``` can be reused later in the method. I do not know if this is better, and this is probably not the right PR for this change, but I am interested in your opinion and that of @jodygarnett.